xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,423 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoStreamSelectHelper.h"
+
+#include "FileItem.h"
+#include "FileItemList.h"
+#include "ServiceBroker.h"
+#include "application/ApplicationComponents.h"
+#include "application/ApplicationPlayer.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "utils/LangCodeExpander.h"
+#include "utils/StreamDetails.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+namespace
+{
+constexpr int FILEITEM_INDEX_DISABLE = -2; // FileItem index value to disable the stream
+constexpr int FILEITEM_INDEX_NONE = -1; // FileItem index value for "none" stream
+
+// \brief Make a FileItem entry with "Disable" label, allow to disable a stream
+CFileItemPtr MakeFileItemDisable(const bool isSelected)
+{
+  CFileItemPtr fileItem = std::make_shared<CFileItem>(g_localizeStrings.Get(24021));
```

Comment:
`const auto fileItem{std::make_shared...`- shorter and more readable and `const` - here and at other places in this file.

Reply:
here i cant use "const auto fileItem"
it will result that return std::shared_ptr<const CFileItem>
and dont work with CFileItemList uses

Reply:
Then you can make it work by not using `auto` but `const std::shared_ptr<CFileItem> fileitem`.

Reply:
oh ok seem works also auto, i just didn't do the same with the method return value